src/util/strencodings.cpp

```diff
@@ -526,3 +527,63 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, const uint64_t default_multiplier)
+{
+    if (str.empty()) {
+        return std::nullopt;
+    }
+
+    using opt_uint64 = std::optional<uint64_t>;
+
+    uint64_t nBytes{0};
+    uint64_t multiplier{default_multiplier};
+    auto unit = str.back();
+    auto v = str;
+
+    v.pop_back();
+
+    switch (unit) {
+        case 'k':
+            multiplier = 1000;
+            break;
+        case 'K':
+            multiplier = 1024;
+            break;
+        case 'm':
+            multiplier = 1000 * 1000;
+            break;
+        case 'M':
+            multiplier = 1ULL << 20;
+            break;
+        case 'g':
+            multiplier = 1000ULL * 1000 * 1000;
+            break;
+        case 'G':
+            multiplier = 1ULL << 30;
+            break;
+        case 't':
+            multiplier = 1000ULL * 1000 * 1000 * 1000;
+            break;
+        case 'T':
+            multiplier = 1ULL << 40;
+            break;
+        default:
+            // unit should be digit, parse original str
+            if (!ParseUInt64(str, &nBytes)) {
+                return std::nullopt;
+            }
+            //check overflow and use default_multiplier
+            return default_multiplier == 0 ? opt_uint64{nBytes} : nBytes > std::numeric_limits<uint64_t>::max() / default_multiplier ? std::nullopt : opt_uint64{nBytes * default_multiplier};
+    }
+
+    if (!ParseUInt64(v, &nBytes)) {
+        return std::nullopt;
+    }
+
+    if (multiplier == 0) {
+        return opt_uint64{nBytes};
+    }
+    // check overflow
+    return nBytes > std::numeric_limits<uint64_t>::max() / multiplier ? std::nullopt : opt_uint64{nBytes * multiplier};
```

Comment:
In commit "util: ParseByteUnits - Parse a string with suffix unit [k|K] [m|M] [g|G] [t|T]" (c5d4a59cf91f54a2027851b32cf83cfcb04cd210)

There's repetition and unneeded variables in this function. Also `nBytes` holds the number before multiplication instead of the actual number of bytes, and the name doesn't follow current coding convention. Would suggest an overall cleanup here like:

```diff
diff --git a/src/util/strencodings.cpp b/src/util/strencodings.cpp
index c490cc7f3e1..0075e69c5e7 100644
--- a/src/util/strencodings.cpp
+++ b/src/util/strencodings.cpp
@@ -530,19 +530,8 @@ std::string HexStr(const Span<const uint8_t> s)
 
 std::optional<uint64_t> ParseByteUnits(const std::string& str, const uint64_t default_multiplier)
 {
-    if (str.empty()) {
-        return std::nullopt;
-    }
-
-    using opt_uint64 = std::optional<uint64_t>;
-
-    uint64_t nBytes{0};
-    uint64_t multiplier{default_multiplier};
-    auto unit = str.back();
-    auto v = str;
-
-    v.pop_back();
-
+    auto multiplier = default_multiplier ? default_multiplier : 1;
+    char unit = str.empty() ? 0 : str.back();
     switch (unit) {
         case 'k':
             multiplier = 1000;
@@ -569,21 +558,13 @@ std::optional<uint64_t> ParseByteUnits(const std::string& str, const uint64_t de
             multiplier = 1ULL << 40;
             break;
         default:
-            // unit should be digit, parse original str
-            if (!ParseUInt64(str, &nBytes)) {
-                return std::nullopt;
-            }
-            //check overflow and use default_multiplier
-            return default_multiplier == 0 ? opt_uint64{nBytes} : nBytes > std::numeric_limits<uint64_t>::max() / default_multiplier ? std::nullopt : opt_uint64{nBytes * default_multiplier};
+            unit = 0;
+            break;
     }
-
-    if (!ParseUInt64(v, &nBytes)) {
+    uint64_t num;
+    if (!ParseUInt64(unit ? str.substr(0, str.size()-1) : str, &num) ||
+        num > std::numeric_limits<uint64_t>::max() / multiplier) {
         return std::nullopt;
     }
-
-    if (multiplier == 0) {
-        return opt_uint64{nBytes};
-    }
-    // check overflow
-    return nBytes > std::numeric_limits<uint64_t>::max() / multiplier ? std::nullopt : opt_uint64{nBytes * multiplier};
+    return num * multiplier;
 }

```

Reply:
Great suggestion, code now looks cleaner.